File: server/FitMate.Core/JsonModels/TrainingProfiles/SaveTrainingProfileRequest.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.TrainingProfiles;

public class SaveTrainingProfileRequest
{
    public TrainingGoal Goal { get; set; }
    public TrainingExperienceLevel ExperienceLevel { get; set; }
    public int PreferredTrainingDaysPerWeek { get; set; }
    public int? PreferredWorkoutDurationMinutes { get; set; }
    public WeightUnit WeightUnit { get; set; } = WeightUnit.Kg;
    public List<string> AvailableEquipment { get; set; } = [];
    public List<DayOfWeek> PreferredTrainingDays { get; set; } = [];
    public string? ExerciseRestrictions { get; set; }
    public string? AdditionalPreferences { get; set; }
    public bool AllowAiPersonalization { get; set; } = true;
}

File: client/src/types/JsonModels/TrainingProfiles/SaveTrainingProfileRequest.ts
import type { DayOfWeek } from "../Enums/DayOfWeek";
import type { TrainingExperienceLevel } from "../Enums/TrainingExperienceLevel";
import type { TrainingGoal } from "../Enums/TrainingGoal";
import type { WeightUnit } from "../Enums/WeightUnit";

export interface SaveTrainingProfileRequest
{
	goal: TrainingGoal;
	experienceLevel: TrainingExperienceLevel;
	preferredTrainingDaysPerWeek: number;
	preferredWorkoutDurationMinutes?: number;
	weightUnit: WeightUnit;
	availableEquipment: string[];
	preferredTrainingDays: DayOfWeek[];
	exerciseRestrictions?: string;
	additionalPreferences?: string;
	allowAIPersonalization: boolean;
}
